docs/troubleshooting.md

```diff
@@ -0,0 +1,127 @@
+
+```
+$ oc get packagemanifests -n openshift-marketplace
+
+NAME                             CATALOG               AGE
+packageserver                    OLM Operators         3h31m
+
+```
+
+If no `packagemanifests` were created, check the logs for the `package-server` pods.
+
+```
+$ oc get pods -n openshift-operator-lifecycle-manager -l app=packageserver
+
+NAME                                READY     STATUS    RESTARTS   AGE
+packageserver-6c664f6d76-jj24b      1/1       Running   0          3h54m
+packageserver-6c664f6d76-kzpd9      1/1       Running   0          3h53m
+
+$ oc logs packageserver-6c664f6d76-jj24b -n openshift-operator-lifecycle-manager
+$ oc logs packageserver-6c664f6d76-kzpd9 -n openshift-operator-lifecycle-manager
+
+```
+
+The logs for the `package-server` pods should contain information about why it might have failed to detect the presence of the `CatalogSources`. If the `package-server` pod does not show up or is crashing (anything other than healthy), try killing the pod or editing the deployment you see when you `oc get deployments -n openshift-operator-lifecycle-manager`.
+
+If everything seems healthy, and still no packages show up in the UI, it could be a browser issue. Check the the browser console to see the logs for possible errors. At any point in the steps above if the error looks too complicated to debug, please reach out to the #forum-operators channel in the [upstream Kubernetes slack instance](https://slack.k8s.io/) or the [google group for Operator-Framework](https://groups.google.com/forum/#!forum/operator-framework). 
+
+
+
+## OperatorSource failing to download from datastore
+
+If an OperatorSource fails to reconcile for any reason (i.e operators in that particular OperatorSource fail to show up in the UI), it might be helpful first inspect the `STATUS` and the `MESSAGE` of the OperatorSource with `oc get opsrc -n openshift-marketplace`. You can also inspect the log messages of the marketplace-operator for detailed information about the OperatorSource. 
+
+Note: A common reason for an OperatorSource to not reconcile correctly has been the use of the wrong [end point](https://github.com/operator-framework/operator-marketplace/blob/master/deploy/examples/community.operatorsource.cr.yaml#L8). Please make sure that the end point is specified as `https://quay.io/cnr`.
+
+## OperatorSource `MESSAGE` reads unknown error (status 404)
+
+Some of the common reasons we have encountered that caused this `MESSAGE` are quay.io being unavailable or the wrong end point being listed in the OperatorSource. If the steps listed [here](#operatorSource-failing-to-download-from-datastore) does not help debug the problem, please reach out to the #forum-operators channel in the [upstream Kubernetes slack instance](https://slack.k8s.io/) or the [google group for Operator-Framework](https://groups.google.com/forum/#!forum/operator-framework). 
+
+## Cannot uninstall operator from the CLI in a clean way. 
```

Comment:
How is this helping with troubleshooting? You are just giving a gist of uninstalling an operator. I suggest removing this section.